Renamed `sampled_at` timestamps to `origin_time`

tiyash-basu-frequenz opened the pull request:
closes #326

tiyash-basu-frequenz committed:
Renamed `sampled_at` timestamps for states to `origin_time`

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

thomas-nicolai-frequenz commented on proto/frequenz/api/common/v1/metrics/metrics.proto:
```diff
@@ -186,7 +186,7 @@ enum Metric {
 //     command, clients need to request current values within the bounds.
 message MetricSample {
   // The UTC timestamp of when the metric was sampled.
```
I would probably say when the metric was collected instead of "sampled".

thomas-nicolai-frequenz replied:
Another suggestion: `// The UTC time when the metric was originally captured.`

thomas-nicolai-frequenz commented on proto/frequenz/api/common/v1/metrics/metrics.proto:
```diff
@@ -186,7 +186,7 @@ enum Metric {
-  google.protobuf.Timestamp sampled_at = 1;
+  google.protobuf.Timestamp origin_time = 1;
 
   // The metric that was sampled.
```
collected?

tiyash-basu-frequenz replied:
As discussed, I updated it to `sample_time`.

thomas-nicolai-frequenz commented on proto/frequenz/api/common/v1/microgrid/electrical_components/electrical_components.proto:
```diff
@@ -476,7 +476,7 @@ message ElectricalComponentDiagnostic {
 // Representation of a component state, warnings, and errors.
 message ElectricalComponentStateSnapshot {
   // The time at which the state was sampled.
```
collected here too?

thomas-nicolai-frequenz replied:
Another suggestion: // The UTC time when the state was originally captured.

tiyash-basu-frequenz replied:
As discussed, I updated it to `sample_time`.

I also updated the documentation.

llucax reviewed (dismissed):
I'm surprised about `origin_time`, I thought and feel for a **sample**, `sample_time` feels more natural and obvious, but I guess this was already discussed and decided on, so I will approve.

Nevertheless, I will disable auto-merge just in case to be extra careful.

thomas-nicolai-frequenz commented on proto/frequenz/api/common/v1/metrics/metrics.proto:
```diff
@@ -186,7 +186,7 @@ enum Metric {
 message MetricSample {
   // The UTC timestamp of when the metric was sampled.
-  google.protobuf.Timestamp sampled_at = 1;
+  google.protobuf.Timestamp origin_time = 1;
```
I think this should stay sampled_at because its a metric. Sampled just doesn't makes sense in the context of states.

tiyash-basu-frequenz committed:
Rename `sampled_at` to `sample_time` in metrics and sensor metrics

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

tiyash-basu-frequenz committed:
Rename snapshot_time to origin_time in communication component state

This commit renames the field `snapshot_time` to `origin_time` in the
`CommunicationComponentStateSnapshot` message. This makes the field name
consistent across `ElectricalComponentStateSnapshot` and
`SensorStateSnapshot` messages.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

tiyash-basu-frequenz commented:
>I'm surprised about origin_time, I thought and feel for a sample, sample_time feels more natural and obvious, but I guess this was already discussed and decided on

That was my understanding as well. But it looks like the decision has been changed now.

tiyash-basu-frequenz commented:
I also noticed that the timestamp field for communication component states was `snapshot_time` I have updated it to `origin_time`.

tiyash-basu-frequenz committed:
Update documentation for `origin_time` field in proto files

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

# Conflicts:
#	proto/frequenz/api/common/v1/microgrid/electrical_components/electrical_components.proto

thomas-nicolai-frequenz reviewed (approved):
LGTM
